teuthology/suite/run.py

```diff
@@ -564,39 +569,102 @@ def check_num_jobs(self, jobs_to_schedule):
         if threshold and jobs_to_schedule > threshold:
             util.schedule_fail(msg, dry_run=self.args.dry_run)
 
-    def schedule_suite(self):
+    def descr_to_yamls(self, descriptions, suites_path):
         """
-        Schedule the suite-run. Returns the number of jobs scheduled.
+        Function converts description of the run into sequence of .yaml files.
+        Input "descriptions" is the value of "--descr" parameter;
+        "description" consists of comma-separated list of run description
```

Comment:
I think you meant plural form "descriptions" instead of singular "description". I suggest to move the splitting a description parameter out of the method, so you can use it like:
```
yamls = yamls_from_descr(self.args.descr.split(","), suites_path)
```

The descriptions for the "suites_path", however it is self description, and because of English, rename to "suite_path".